R/data.R

```diff
@@ -67,14 +67,10 @@
 #'   `vec_restore(vec_data(x), x)` will always yield `x`.
 #' @export
 vec_data <- function(x) {
-  if (!is_vector(x)) {
-    return(x)
-  }
+  vec_assert(x)
+  x <- vec_proxy(x)
 
-  # TODO: implement with ALTREP to avoid making a copy
-  if (is_record(x)) {
-    x <- vec_set_attributes(x, list(names = fields(x)))
-  } else if (has_dim(x)) {
+  if (has_dim(x)) {
```

Comment:
What allows you to return the `is_record()` branch? The `names()` of a record shouldn't return the field names.

Reply:
Oh because you're proxying it first, duh.